docs/root/configuration/upstream/cluster_manager/cds.rst

```diff
@@ -18,3 +18,40 @@ Statistics
 ----------
 
 CDS has a :ref:`statistics <subscription_statistics>` tree rooted at *cluster_manager.cds.*
+
+On-demand CDS
+-------------
+
+Similar to VHDS on-demand feature in terms of hosts, the on-demand CDS API is an additional API
+that Envoy will call to dynamically fetch upstream clusters which Envoy interested in spontaneously.
+
+By default in CDS, all cluster configurations are sent to every Envoy instance in the mesh. The
+delta CDS provides the ability that the xDS management server can send incremental CDS to the Envoy
+instance, but Envoy instance can not feedback upstream clusters it interested in spontaneously, in
+other words, Envoy instance only can receive the CDS passively.
+
+In order to fix this issue, on-demand CDS uses the delta xDS protocol to allow a cluster configuration
+to be subscribed to and the necessary cluster configuration to be requested as needed. Instead
+of sending all cluster configuration or cluster configuration the Envoy instance aren't interested
+in, using on-demand CDS will allow an Envoy instance to subscribe and unsubscribe from a list of
+cluster configurations stored internally in the xDS management server. The xDS management server
+will monitor the list and use it to filter the configuration sent to an individual Envoy instance
+to only contain the subscribed cluster configurations.
```

Comment:
Is this true? Let's say we have an Envoy and it has a CDS initially with {X, Y}. An extension notices that Z is missing and asks for Z. Is the management server now _only_ returning {Z} or is it {X, Y, Z} (via delta protocol). I think what we're doing with on-demand is augmenting the base set of clusters, not going pure delta subscription.

Reply:
the management server will only return {Z}, but actually it depends the implementation of management server, management server can also return {X, Y, Z}, but in the scene we expected before, the management server will only return {Z}.

Reply:
I think we need to describe two different things here:
1. What the management server thinks the Envoy is subscribed to. Presumably {X, Y, Z}.
2. What the management server puts on the wire as part of the response delta update, this is just {Z}, but it could validly be any superset.